src/reactions/ReactionsReader.ts

```diff
@@ -0,0 +1,336 @@
+/*
+Copyright 2024 New Vector Ltd.
+
+SPDX-License-Identifier: AGPL-3.0-only
+Please see LICENSE in the repository root for full details.
+*/
+
+import {
+  type CallMembership,
+  MatrixRTCSessionEvent,
+  type MatrixRTCSession,
+} from "matrix-js-sdk/src/matrixrtc";
+import { logger } from "matrix-js-sdk/src/logger";
+import { type MatrixEvent, MatrixEventEvent } from "matrix-js-sdk/src/matrix";
+import { type ReactionEventContent } from "matrix-js-sdk/src/types";
+import {
+  RelationType,
+  EventType,
+  RoomEvent as MatrixRoomEvent,
+} from "matrix-js-sdk/src/matrix";
+import { BehaviorSubject, delay } from "rxjs";
+
+import {
+  ElementCallReactionEventType,
+  type ECallReactionEventContent,
+  GenericReaction,
+  ReactionSet,
+  type RaisedHandInfo,
+  type ReactionInfo,
+} from ".";
+
+export const REACTION_ACTIVE_TIME_MS = 3000;
+
+/**
+ * Listens for reactions from a RTCSession and populates subjects
+ * for consumption by the CallViewModel.
+ * @param rtcSession
+ */
+export class ReactionsReader {
+  private readonly raisedHandsSubject$ = new BehaviorSubject<
+    Record<string, RaisedHandInfo>
+  >({});
+  private readonly reactionsSubject$ = new BehaviorSubject<
+    Record<string, ReactionInfo>
+  >({});
+
+  /**
+   * The latest set of raised hands.
+   */
+  public readonly raisedHands$ = this.raisedHandsSubject$.asObservable();
+
+  /**
+   * The latest set of reactions.
+   */
+  public readonly reactions$ = this.reactionsSubject$.asObservable();
+
+  public constructor(private readonly rtcSession: MatrixRTCSession) {
+    // Hide reactions after a given time.
+    this.reactionsSubject$
+      .pipe(delay(REACTION_ACTIVE_TIME_MS))
+      .subscribe((reactions) => {
```

Comment:
This subscription should be bound to some `ObservableScope` so that it doesn't run forever